FWCore/Framework/interface/PathsAndConsumesOfModules.h

```diff
@@ -54,10 +64,20 @@ namespace edm {
     std::vector<ModuleDescription const*> const& doModulesWhoseProductsAreConsumedBy(
         unsigned int moduleID, BranchType branchType) const override;
 
+    std::vector<eventsetup::ComponentDescription const*> const& doESModulesWhoseProductsAreConsumedBy(
+        unsigned int moduleID, Transition) const override;
+
     std::vector<ConsumesInfo> doConsumesInfo(unsigned int moduleID) const override;
+    std::vector<EventSetupConsumesInfo> doEventSetupConsumesInfo(unsigned int moduleID) const override;
 
     unsigned int doLargestModuleID() const override;
 
+    std::vector<eventsetup::ESProductResolverProvider const*> const& doAllESProductResolverProviders() const override;
```

Comment:
I don't like exposing even via forward reference `eventsetup::ESProductResolverProvider` to everything that deals with a PathsAndConsumesOfModulesBase.

Reply:
I removed this dependence also from the return value to this function.